Src/AutoFixture/RandomNumericSequenceGenerator.cs

```diff
@@ -104,54 +105,18 @@ private static void ValidateThatLimitsAreStrictlyAscending(long[] limits)
 
         private object CreateRandom(Type request)
         {
-            switch (Type.GetTypeCode(request))
+            if(!request.IsNumeric())
             {
-                case TypeCode.Byte:
-                    return (byte)
-                        this.GetNextRandom();
-
-                case TypeCode.Decimal:
-                    return (decimal)
-                        this.GetNextRandom();
-
-                case TypeCode.Double:
-                    return (double)
-                        this.GetNextRandom();
-
-                case TypeCode.Int16:
-                    return (short)
-                        this.GetNextRandom();
-
-                case TypeCode.Int32:
-                    return (int)
-                        this.GetNextRandom();
-
-                case TypeCode.Int64:
-                    return
-                        this.GetNextRandom();
-
-                case TypeCode.SByte:
-                    return (sbyte)
-                        this.GetNextRandom();
-
-                case TypeCode.Single:
-                    return (float)
-                        this.GetNextRandom();
-
-                case TypeCode.UInt16:
-                    return (ushort)
-                        this.GetNextRandom();
-
-                case TypeCode.UInt32:
-                    return (uint)
-                        this.GetNextRandom();
-
-                case TypeCode.UInt64:
-                    return (ulong)
-                        this.GetNextRandom();
-
-                default:
-                    return new NoSpecimen();
+                return new NoSpecimen();
+            }
+            var nextRandom = GetNextRandom();
+            try
+            {
+                return Convert.ChangeType(nextRandom, request, CultureInfo.InvariantCulture);
+            }
+            catch(OverflowException)
+            {
+                return Convert.ChangeType(0, request, CultureInfo.InvariantCulture);

```

Comment:
> for reasons I haven't investigated

Oh, I see now that the first range point was changed to `SByte.MaxValue`, which is 127, so that explains that...